### PR TITLE
ISPN-11685 Update to RxJava 3.0

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-11685

### CODE REVIEW

**Comment by wburns** [comment]
The latter commit is more interesting as I removed our usage of internal rxjava classes and replaced some code with new, especially InnerPublisherSubscription.

File: core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java [hunk 1]
```diff
@@ -206,6 +206,8 @@ public void start() {
       // This excludes the keys from the various address targets
       Map<Address, Set<K>> keyTargets = determineKeyTargets(topology, keysToInclude, localAddress, segments, ctx);
 
+      log.fatal("keyTargets is: " + keyTargets);
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java:209]: Should these log.fatal statements still be here? :fearful:

File: core/src/test/java/org/infinispan/stream/DistributedParallelStreamTest.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  * Verifies stream tests work when stream is parallel with parallel distribution
  */
 @Test(groups = "functional", testName = "streams.DistributedParallelStreamTest")
-@InCacheMode({ CacheMode.DIST_SYNC, CacheMode.SCATTERED_SYNC })
+@InCacheMode({ CacheMode.DIST_SYNC/*, CacheMode.SCATTERED_SYNC*/ })
```
Comment (ryanemerson) [inline core/src/test/java/org/infinispan/stream/DistributedParallelStreamTest.java:13]: Uncomment?

File: core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java (hunk 1 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java:209]: Whoops forgot to remove it :)

File: core/src/test/java/org/infinispan/stream/DistributedParallelStreamTest.java (hunk 2 above)
Comment (wburns) [reply core/src/test/java/org/infinispan/stream/DistributedParallelStreamTest.java:13]: Same, forgot to revert :+1:

**Comment by wburns** [comment]
Updated to remove and added back in javadoc comments I accidentally removed.

**Comment by ryanemerson** [comment]
Thanks @wburns

**Pull request merged**